training/train.py

```diff
@@ -354,59 +319,141 @@ def get_deployed_model_version(yaml_data, deployment_prefix):
                 model_version = env.get('parameters', {}).get('MODEL_VERSION', '')
                 break
     else:
-        _logger.error('ERROR - Invalid deployment prefix: "{}", supported values: {}'.format(
+        raise Exception('Invalid deployment prefix: "{}", supported values: {}'.format(
             deployment_prefix, list(DEPLOYMENT_PREFIX_ENVIRONMENT_NAME_MAP.keys())))
 
     _logger.info('Model version: {} for deployment prefix: {}'.format(
         model_version, deployment_prefix))
     return model_version
 
 
+def update_yaml_data(yaml_data, deployment_prefix, model_version, description):
```

Comment:
Can you think of defining a property to contains comments?
e.g.
```
Description: >
param: val
...
```
This will enable you to keep using yaml parser without adhoc parsing logics

Reply:
I will look into this, to see if we can use yaml package instead of custom code.

Reply:
This is done with new `ruamel.yaml` package.